### PR TITLE
Http to https redirect

### PR DESCRIPTION
**Pull request by bkuhl**
* nginx config is now generated allowing it to utilize environment variables
 * `production` and `staging` environments are redirected to https
 * supports ssl redirecting sitting behind HAProxy and ELB
 * `/healthcheck` endpoint is not redirected to https - contributed by @millidavids

### CODE REVIEW

**Comment by bkuhl** [comment]
📟

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 /node_modules
 /public/storage
 /public/css
+#infrastructure/nginx/default.conf
```
Comment (jlaswell) [inline .gitignore]: Should this stay commented, or just be removed?

Comment (bkuhl) [inline .gitignore]: Removed

Comment (bkuhl) [inline .gitignore]: Actually, let me rework this to make it more clear what's going on.

Comment (bkuhl) [inline .gitignore]: This has been removed

**Comment by millidavids** [comment]
+1

**Pull request merged**